### PR TITLE
docs: how to create a new aks-engine release

### LINKED ISSUE #577: Document release procedure
**Is this a request for help?**:
NO

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
FEATURE REQUEST

---

We need a simple, tested doc on how to release a new version of aks-engine. It's not that complicated but there are several steps that are easy to forget, including:

- make sure to `git clean -fdx` before `make dist` or the `aks-engine version` command shows the binaries as "dirty"
- update the `gofish` index with a PR to fishworks/fish-food
- update the `brew` index with a PR to Azure/homebrew-aks-engine

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

Documents the mechanical process of releasing aks-engine and updates other sections.

**Issue Fixed**:

Fixes #577

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
FYI the first paragraph content under `## aks-engine Releases Each Month` is not strictly true

File: docs/community/release-checklist.md [hunk 1]
```diff
@@ -47,54 +47,103 @@ In general, changes to anything a user might reasonably link to, customize, or i
...
 to a patch or to a minor release will not break anything.
 
-## How to Release aks-engine
+## Creating a New Release
 
-This section leads a maintainer through creating an aks-engine release.
+Let's go through the process of creating a new release of [aks-engine][].
 
-### Step 1: Assemble Master Changelog
+We will use **v0.32.3** as an example herein. You should replace this with the new version you're releasing.
```
Comment (CecileRobertMichon) [inline docs/community/release-checklist.md]: would it be easier to add a bash var `VERSION=v0.32.3` and then use that in the rest of the commands so they can be easily copied/pasted?

Comment (mboersma) [reply docs/community/release-checklist.md]: Good idea—done.

**Comment by mboersma** [comment]
> `Releases Each Month` is not strictly true

I updated that section to reflect reality.

**Pull request merged**